README.md

```diff
@@ -10,19 +10,31 @@ Staticman is a Node.js application that receives user-generated content and uplo
 
 It consists of a small web service that handles the `POST` requests from your forms, runs various forms of validation and manipulation defined by you and finally pushes them to your repository as data files. You can choose to enable moderation, which means files will be pushed to a separate branch and a pull request will be created for your approval, or disable it completely, meaning that files will be pushed to the main branch automatically.
 
-You can download and run the Staticman API on your own infrastructure. The easiest way to get a personal Staticman API instance up and running is to use the free tier of Heroku. If deploying to Heroku you can simply click the button below and enter your config variables directly into Heroku as environment variables.
-
-[![Deploy](https://www.herokucdn.com/deploy/button.svg)](https://heroku.com/deploy)
-
 ## Requirements
 
 - Node.js 8.11.3+
 - npm
 - A [personal access token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) for the GitHub and/or GitLab account you want to run Staticman with
 - An RSA key in PEM format
 
+## Setting up your free Heroku instance (RECOMMENDED)
+
+- create a free account on Heroku and install the [Heroku CLI](https://devcenter.heroku.com/articles/heroku-cli#download-and-install) on your workstation
```

Comment:
The button is not needed, `create-heroku` Make target takes care of creating the app.

Even the bot GitHub account is not strictly needed. I'm not using it for example and I find it overkill for most users. What's exactly the threat scenario we're trying to overcome forcing the user to create a dedicated bot account? For how I see it, it might be needed in some high end setups, so it could be mentioned in some "advanced" kind of section.

I'd say the way to go is to use Oauth or whatever GitHub provides ad an auth mechanism to give read/write access to a selected list of repositories (like Netlify or other apps do for example).

Reply:
I believe the most common threat with the present approach (github token for account) is that a compromised GitHub token can be used to access all repos unrestricted under the main account.

Reply:
I understand the threat but I believe the user should make that choice instead of being forced to:
1. create a new dedicated email address for Staticman
1. create a new dedicated GH user for Staticman
1. create a token
1. authorized the user into any GH project where Staticman is needed
1. maybe I'm missing something, but even if not it's already way too complex for my taste

Personally I'm ok with the lower security trade off in this case, I trust enough Heroku to keep the env variable safe and I trust myself to keep them safe as well, e.g. in the unlikely case some adversary can get access to this secret I'll have to face way more serious problems than a stolen GH token 😆

I'm not saying securing a Staticman setup is not important though, my argument is:
- the decrease of UX experience is not worth the benefit for most users
- right now, we should give the user the chance to choose between the two
- in the future, we should make it possible to have it both secure AND easy to setup, for example using some Oauth workflows

Reply:
The button leave user deploying without command line, while make command require an working *nix environment with gnu make installed. Maybe the UX experience of *nix user is improved, but it's a disaster for all others.

I understand you want everything done in single place, but it would be better splitting into different sections:

- Setting up your free Heroku instance
  - With `make` command
  - With `deploy to Heroku` button

Also, should we clarify the `make` solution is bind to which working mode and what scenario is targeting to? It looks like this solution will grant permission on EVERY repo I had access to and will identify as MY SELF. The risk should not be ignored with "RECOMMENDED".

Reply:
Thank you @pingu8007 for reviving the discussion here, I'd be really happy to see some more activity.

> The button leave user deploying without command line, while make command require an working *nix environment with gnu make installed. Maybe the UX experience of *nix user is improved, but it's a disaster for all others.

Good observation, the problem is that the 1-click button is not really working unless you also provide some configuration and that apparently creates a lot of friction and most importantly a lot of bug/requests for help to set it up. Personally I believe this is one of the main reason of friction: https://github.com/eduardoboucas/staticman/issues/368
That said, the question now is: what's the target user? because I'm afraid if someone has zero knowledge of *nix setups it will require quite some more extra effort to fill the gaps and make it usable. I understand the will, it's a very good target, but I see it a bit far in the roadmap. I'd suggest to go for:

1. make the UX good for "power" users
1. make the UX good even for non-techies

> I understand you want everything done in single place, but it would be better splitting into different sections:
> 
>     * Setting up your free Heroku instance
>       
>       * With `make` command
>       * With `deploy to Heroku` button

My goal is to fix the UX, which is currently broken because docs are out of date and setup cumbersome. I'm open to whatever recommendation to get there, it doesn't necessarily have to be in a _single_ place. Your suggestion sounds good though, I'll fix that.

> Also, should we clarify the `make` solution is bind to which working mode and what scenario is targeting to? It looks like this solution will grant permission on EVERY repo I had access to and will identify as MY SELF. The risk should not be ignored with "RECOMMENDED".

Makes sense, the user should be aware of the security implications and make the choice. Again, this should not the be the case with most recent (and undocumented) workflow which lets you connect applications to selected GH repositories, but I honestly don't know if it is implemented correctly or not.

More in general I believe the true problem now lies in the fact that there's no clear governance of the project, hence any kind of development is either very slow or worse discouraged for interested newcomers.

Any recommendation on how to fix that also very welcome!

Reply:
> My goal is to fix the UX, which is currently broken because docs are out of date and setup cumbersome.

I'm starting a [wiki page](https://github.com/eduardoboucas/staticman/wiki/Difference-of-each-working-mode-(GitHub)) to describe different working mode, and I will taking care the [GitHub App] mode. Hope this will help future discussion on which way should we recommend to user. :D

Reply:
> > My goal is to fix the UX, which is currently broken because docs are out of date and setup cumbersome.
> 
> I'm starting a [wiki page](https://github.com/eduardoboucas/staticman/wiki/Difference-of-each-working-mode-(GitHub)) to describe different working mode, and I will taking care the [GitHub App] mode. Hope this will help future discussion on which way should we recommend to user. :D

Awesome!

I have gone through it and modified some details. Please let me know if this looks okay. 

Once it is complete, I can make a flow chart/decision tree that will help users decide which path/mode to use (hopefully).

Reply:
@pingu8007 @LordAmit awesome indeed, I think it's great so I don't want to slow you down in the wiki effort, I only want to add that I believe it would be best to (at least eventually) have all the user documentation available at https://staticman.net, I provided a patch recently and even though it takes a little bit more effort to have it right (setup Jekyll locally, create PR, etc) consolidating user docs in one easy-to-access place is definitely a bonus to improve the overall UX.